netty/src/main/java/io/grpc/netty/NettyAdaptiveCumulator.java

```diff
@@ -22,10 +22,18 @@
 import io.netty.buffer.ByteBufAllocator;
 import io.netty.buffer.CompositeByteBuf;
 import io.netty.handler.codec.ByteToMessageDecoder.Cumulator;
+import io.netty.util.Version;
 
 class NettyAdaptiveCumulator implements Cumulator {
+  private static final boolean usingPre4_1_111_Netty;
   private final int composeMinSize;
 
+  static {
+    Version version = Version.identify().get("netty-buffer");
+    usingPre4_1_111_Netty =
+        version != null && version.artifactVersion().compareTo("4.1.111.Final") < 0;
```

Comment:
How likely the version would be `4.1.99` ?

Reply:
Good point.  I've updated the logic.